feedthefox/users/views.py

```diff
@@ -1,11 +1,29 @@
+import json
+
 from django.contrib.auth.decorators import login_required
-from django.http import Http404
+from django.http import Http404, HttpResponse
 from django.shortcuts import get_object_or_404, redirect, render
 
-from feedthefox.devices.models import DeviceInfo
+from feedthefox.devices.models import DeviceInfo, validate_imei
 from feedthefox.users import forms
 
 
+def imei(request, imei):
+    if request.is_ajax():
+        response_data = {}
+        try:
+            status = validate_imei(imei)
+            response_data['status'] = 1
+        except:
+            response_data['status'] = 0
+        return HttpResponse(
+            json.dumps(response_data),
+            content_type="application/json"
+        )
+    else:
+        raise Http404
```

Comment:
Instead of raising a 404 error, shouldn't we just display the form errors if exist?